dpsim-villas/src/InterfaceVillasQueueless.cpp

```diff
@@ -0,0 +1,374 @@
+        throw RuntimeError("Unsupported attribute type!");
+      }
+    }
+
+    sample_decref(sample);
+  } catch (const std::exception &) {
+    if (sample)
+      sample_decref(sample);
+
+    throw;
+  }
+  return seqnum;
+}
+
+void InterfaceVillasQueueless::PostStep::execute(Real time, Int timeStepCount) {
+  mIntf.writeToVillas();
+}
+
+void InterfaceVillasQueueless::writeToVillas() {
+  if (mExportAttrsDpsim.size() == 0) {
+    return;
+  }
+  node::Sample *sample = nullptr;
+  Int ret = 0;
+  try {
+    sample = node::sample_alloc(&mSamplePool);
+    if (sample == nullptr) {
+      SPDLOG_LOGGER_ERROR(mLog, "InterfaceVillas could not allocate a new "
+                                "sample! Not sending any data!");
+      return;
+    }
+
+    sample->signals = mNode->getOutputSignals(false);
+
+    for (size_t i = 0; i < mExportAttrsDpsim.size(); i++) {
+      auto attr = std::get<0>(mExportAttrsDpsim[i]);
+      if (attr->getType() == typeid(Real)) {
+        auto attrReal =
+            std::dynamic_pointer_cast<Attribute<Real>>(attr.getPtr());
+        sample->data[i].f = attrReal->get();
+      } else if (attr->getType() == typeid(Int)) {
+        auto attrInt = std::dynamic_pointer_cast<Attribute<Int>>(attr.getPtr());
+        sample->data[i].i = attrInt->get();
+      } else if (attr->getType() == typeid(Bool)) {
+        auto attrBool =
+            std::dynamic_pointer_cast<Attribute<Bool>>(attr.getPtr());
+        sample->data[i].b = attrBool->get();
+      } else if (attr->getType() == typeid(Complex)) {
+        auto attrComplex =
+            std::dynamic_pointer_cast<Attribute<Complex>>(attr.getPtr());
+        sample->data[i].z = std::complex<float>(attrComplex->get().real(),
+                                                attrComplex->get().imag());
+      } else {
+        SPDLOG_LOGGER_ERROR(mLog, "Error: Unsupported attribute type!");
+        throw RuntimeError("Unsupported attribute type!");
+      }
+    }
+
+    sample->length = mExportAttrsDpsim.size();
+    sample->sequence = mSequenceFromDpsim++;
+    sample->flags |= (int)villas::node::SampleFlags::HAS_SEQUENCE;
+    sample->flags |= (int)villas::node::SampleFlags::HAS_DATA;
+    clock_gettime(CLOCK_REALTIME, &sample->ts.origin);
+    sample->flags |= (int)villas::node::SampleFlags::HAS_TS_ORIGIN;
+
+    do {
+      ret = mNode->write(&sample, 1);
+    } while (ret == 0);
+    if (ret < 0)
+      SPDLOG_LOGGER_ERROR(mLog,
+                          "Failed to write samples to InterfaceVillas. Write "
+                          "returned code {}",
+                          ret);
+
+    sample_decref(sample);
+  } catch (const std::exception &) {
+    sample_decref(sample);
+
+    if (ret < 0)
+      SPDLOG_LOGGER_ERROR(
+          mLog,
+          "Failed to write samples to InterfaceVillas. Write returned code {}",
+          ret);
+
+    /* Don't throw here, because we managed to send something */
```

Comment:
Please use C++ style comments
```suggestion
    // Don't throw here, because we managed to send something.
```